parsl/executors/high_throughput/executor.py

```diff
@@ -327,9 +336,25 @@ def initialize_scaling(self):
     def start(self):
         """Create the Interchange process and connect to it.
         """
-        self.outgoing_q = zmq_pipes.TasksOutgoing("127.0.0.1", self.interchange_port_range)
-        self.incoming_q = zmq_pipes.ResultsIncoming("127.0.0.1", self.interchange_port_range)
-        self.command_client = zmq_pipes.CommandClient("127.0.0.1", self.interchange_port_range)
+        if self.encrypted and self.cert_dir is None:
+            logger.debug("Creating CurveZMQ certificates")
+            self.cert_dir = curvezmq.create_certificates(self.run_dir)
+        elif not self.encrypted and self.cert_dir:
+            raise AttributeError(
+                "The certificates directory path attribute (cert_dir) is defined, but the "
+                "encrypted attribute is set to False. You must either change cert_dir to "
+                "None or encrypted to True."
+            )
+
```

Comment:
1. We call `curvezmq.create_certificates` in the `start` method because `run_dir` is usually defined after instantiation.
2. The `encrypted` attribute needs to come from the initializer arguments. Otherwise, users won't have a mechanism to disable encryption.
3. The `elif` block and related exception are not required but could assist with troubleshooting.